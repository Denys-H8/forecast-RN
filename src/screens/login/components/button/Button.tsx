import React from 'react';
import { TouchableOpacity } from 'react-native';

import { styles } from './button.styles';
import { CustomText } from 'src/components';

interface IProps {
  onPress(): void;
  style?: object;
  isDisabled: boolean;
}

const Input: React.FC<IProps> = ({ onPress, style, isDisabled }) => {
  const disableStyle = isDisabled ? { opacity: 0.4 } : {};

  return (
    <TouchableOpacity
      onPress={onPress}
      disabled={isDisabled}
      style={[styles.button, style, disableStyle]}
      testID="Button.TouchableOpacity"
    >
      <CustomText style={styles.text}>LOGIN</CustomText>
    </TouchableOpacity>
  );
};

export default Input;
