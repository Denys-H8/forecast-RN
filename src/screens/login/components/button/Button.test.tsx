import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import Button from './Button';

const styleMock = {
  backgroundColor: 'green',
};

describe('Button ', () => {
  test('render not disabled button', () => {
    const onPressMock = jest.fn();

    const { toJSON, getByTestId } = render(<Button isDisabled={false} style={styleMock} onPress={onPressMock} />);

    fireEvent.press(getByTestId('Button.TouchableOpacity'));

    expect(toJSON()).toMatchSnapshot();
    expect(onPressMock).toHaveBeenCalled();
  });

  test('render disabled button', () => {
    const onPressMock = jest.fn();

    const { toJSON, getByTestId } = render(<Button isDisabled={true} style={styleMock} onPress={onPressMock} />);

    fireEvent.press(getByTestId('Button.TouchableOpacity'));

    expect(toJSON()).toMatchSnapshot();
    expect(onPressMock).not.toHaveBeenCalled();
  });
});
