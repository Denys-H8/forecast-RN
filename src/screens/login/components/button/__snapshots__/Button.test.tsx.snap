// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button  render disabled button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "green",
      "borderRadius": 100,
      "height": 48,
      "justifyContent": "center",
      "opacity": 0.4,
    }
  }
  testID="Button.TouchableOpacity"
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "WorkSans-Regular",
        },
        Object {
          "color": "#fff",
          "fontSize": 16,
          "lineHeight": 18,
        },
      ]
    }
  >
    LOGIN
  </Text>
</View>
`;

exports[`Button  render not disabled button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "green",
      "borderRadius": 100,
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="Button.TouchableOpacity"
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "WorkSans-Regular",
        },
        Object {
          "color": "#fff",
          "fontSize": 16,
          "lineHeight": 18,
        },
      ]
    }
  >
    LOGIN
  </Text>
</View>
`;
